src/osp/Active/ActiveScene.cpp

```diff
@@ -258,6 +258,12 @@ void ActiveScene::draw(entt::entity camera)
     m_renderOrder.call(cameraComp);
 }
 
+void ActiveScene::system_machine_add(std::string const& name,

```

Comment:
Since this function is always emplacing a new item, i would personally take both of the parameters by value (not const& and not &&, just by value) and std::move them into place.

Additionally, consider having this function either return what m_sysMachines.emplace is returning, or at least have it return a bool indicating success / failure, unless you either

1) Know for certain that it'll only be called when no item already exists.
or
2) don't care if the item already exists.